llvm/CMakeLists.txt

```
@@ -6,7 +6,12 @@ set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
 
-set(CMAKE_BUILD_WITH_INSTALL_NAME_DIR ON)
+# Builds with custom install names and installation rpath setups may not work
+# in the build tree. Allow these cases to use CMake's default build tree
+# behavior by setting `LLVM_USE_INSTALL_NAME_DIR_IN_BUILD_TREE` to do this.
+if (NOT LLVM_USE_INSTALL_NAME_DIR_IN_BUILD_TREE)
```

Follow the style of the rest of the CMake code.
```suggestion
if(NOT LLVM_USE_INSTALL_NAME_DIR_IN_BUILD_TREE)
```
Ditto `endif()`.

---

The logic reads inverted to me. It seems `LLVM_USE_INSTALL_NAME_DIR_IN_BUILD_TREE=ON` should mean `CMAKE_BUILD_WITH_INSTALL_NAME_DIR=ON`, not the opposite.

---

Oh, indeed. I think I named it that and then realized that the default behavior needs to remain unchanged. How about `LLVM_NO_INSTALL_NAME_DIR_FOR_BUILD_TREE`?

---

Yes, that looks good!